packages/circuit-ui/components/CheckboxGroup/CheckboxGroup.tsx

```diff
@@ -0,0 +1,214 @@
+/**
+ * Copyright 2022, SumUp Ltd.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  FieldsetHTMLAttributes,
+  InputHTMLAttributes,
+  Ref,
+  forwardRef,
+} from 'react';
+
+import styled from '../../styles/styled';
+import { typography } from '../../styles/style-mixins';
+import { uniqueId } from '../../util/id';
+import { Checkbox, CheckboxProps } from '../Checkbox/Checkbox';
+import {
+  FieldWrapper,
+  FieldLabelText,
+  FieldValidationHint,
+} from '../FieldAtoms';
+import { AccessibilityError } from '../../util/errors';
+
+export interface CheckboxGroupProps
+  extends Omit<FieldsetHTMLAttributes<HTMLFieldSetElement>, 'onChange'> {
+  /**
+   * A collection of available options. Each option must have at least a label and a value
+   * for the respective Checkbox.
+   */
+  options: (Omit<CheckboxProps, 'onChange'> & {
+    label: string;
+  })[];
+  /**
+   * The values of the Checkboxes that are checked by default.
+   */
+  value?: CheckboxProps['value'][];
+  /**
+   * A callback that is called when any of the checkboxes change their values.
+   * Passed on to the Checkboxes.
+   */
+  onChange: CheckboxProps['onChange'];
+  /**
+   * A description of the selector group.
+   */
+  label: string;
+  /**
+   * Label to indicate that the input group is optional. Only displayed when the
+   * `required` prop is falsy.
+   */
+  optionalLabel?: string;
+  /**
+   * The ref to the HTML DOM element.
+   */
+  ref?: Ref<HTMLFieldSetElement>;
+  /**
+   * Makes the input group required.
+   */
+  required?: InputHTMLAttributes<HTMLInputElement>['required'];
```

Comment:
Yes that's a very valid point! It's better to be able to specify which checkboxes are required. I've included the `required` prop for each `option`, and removed the `optionalLabel`. 
 I guess we wouldn't need a `optionalLabel` for an optional checkbox (i.e., `required` is falsy)? Or do you think we should have a label for the group in case some of the checkboxes are optional?

Reply:
I can think of two scenarios of required checkboxes in a group:

1. At least _x_ amount of options need to be checked. Unfortunately, [HTML5 does not directly support](https://www.w3.org/html/wg/tracker/issues/111?changelog) requiring only one/at least one checkbox to be checked in a checkbox group, so this needs to be validated with JavaScript by the developers.
2. A specific subset of options needs to be checked. This can be achieved by passing the `required` attribute to these specific options. However, this seems like an anti-pattern: it would be more user-friendly to precheck the options and disable them to indicate to the users that these options cannot/should not be changed.

In other words, I don't think the `required` is useful in the context of a checkbox group.

Reply:
I've changed my mind: we should allow the `required` prop, not for semantic reasons, but to toggle the `optionalLabel`.